docs/source/app-dev/command-deduplication.rst

```diff
@@ -0,0 +1,352 @@
+.. Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+.. _command-deduplication:
+
+Command deduplication
+#####################
+
+The interaction of a Daml application with the ledger is inherently asynchronous: applications send commands to the ledger, and some time later they see the effect of that command on the ledger.
+Many things can fail during this time window: the application can crash, the participant node can crash, messages can be lost on the network, or the ledger may be just slow to respond due to a high load.
+If you want to make sure that an intended ledger change is not executed twice, your application needs to robustly handle these failure scenarios.
+This guide covers the following topics:
+
+#. How command deduplication works.
+
+#. How applications can effectively use the command deduplication.
+
+.. _command-dedup-workings:
+
+How command deduplication works
+*******************************
+
+The following fields in a command submissions are relevant for command deduplication.
+The first three form the :ref:`change ID <change-id>` that identifies the intended ledger change.
+
+#. The union of :ref:`party <com.daml.ledger.api.v1.Commands.party>` and :ref:`act_as <com.daml.ledger.api.v1.Commands.act_as>` define the submitting parties.
+  
+#. The :ref:`application ID <com.daml.ledger.api.v1.Commands.application_id>` identifies the application that submits the command.
```

Comment:
This should probably be an unordered list. Generally reserve ordered lists for action steps.

Reply:
I chose an ordered list because the paragraph before referred to the first three items. But you're right, numbering the items doesn't add any clarity. Changed to unordered.